Firestore/core/src/model/transform_operation.cc

```diff
@@ -251,8 +251,12 @@ FieldValue ArrayTransform::Rep::Apply(
       }
     } else {
       HARD_ASSERT(type_ == Type::ArrayRemove);
-      if (pos != result.end()) {
-        result.erase(pos);
+      for (int i = 0; i < static_cast<int>(result.size());) {
+        if (type_ == Type::ArrayRemove && element == result.at(i)) {
```

Comment:
The first part of this is always true, isn't it?

Reply:
Oops, forgot to remove.